lib/models/addon.js

```diff
@@ -131,8 +131,9 @@ var Addon = CoreObject.extend({
   hintingEnabled: function() {
     var isProduction = process.env.EMBER_ENV === 'production';
     var testsEnabledDefault = process.env.EMBER_CLI_TEST_COMMAND || !isProduction;
+    var hintingDisabled = this.app && this.app.options && this.app.options.hinting === false;
```

Comment:
Pedantic, but can we call this `explicitlyDisabled`? Having `hintingEnabled` and `hintingDisabled` in the same method made me triple-check.